atf_parallels/loop.sh

```diff
@@ -13,12 +13,32 @@ _lockfile=.lock
 _image_name=atf_worker
 _container_name=$(basename $_tmpdirname)
 
+_loop_device=""
+_loop_device_file="loop_device_file_"$_container_name".img"
+_loop_device_write_rate=0.25mb
+
 ####################################################################
 #   The following code has to be run in several different processes
 #       i.e. `screen -d -m 'script_to_run.sh'`
 ####################################################################
 
+function create_loop_device {
+    echo "Create Loop Device"
+    dd if=/dev/zero of=$_loop_device_file bs=1024k count=10
+    mkfs -b 1024 $_loop_device_file
+    losetup -fP $_loop_device_file
+    _loop_device=$(losetup -l -n -j $_loop_device_file | awk '{print $1}')
+    echo "Loop Device:" $_loop_device
+}
+
+function delete_loop_device {
+    echo "Delete Loop Device"
+    losetup -d $_loop_device
+    rm -f $_loop_device_file
+}
+
 function docker_run {
+    create_loop_device
```

Comment:
how slow drive is used? is it used for all ATF test scripts? or is it only used by a small set of test scripts related to "Configurable time before shutdown".  Will adding a slow drive slow down running the atf test suites?

Reply:
@yang1070 Please notice this specific ATF branch mostly required for scripts related to SDL-0117 feature.
It is not recommended to use it for other scripts. Even though other scripts wouldn't be slow down however there could be an issues with `loop` devices which would be created and deleted for each script run.

This and some other issues are the reasons why this ATF branch is not part of delivery and should only be used for testing the feature.

Reply:
@dboltovskyi are you saying that this change will not be merged into the release branch even if after the review and the SDL-0117 feature is completed ? otherwise it will impact the performance or stability of existing ATF tests?
is it possible to add some parameters to control the docker run so that for some test cases a slow disk is set up and for most test cases no slow disk created?

Reply:
@@dboltovskyi  my concern is this change will cause the all other atf test cases takes a longer time to run

Reply:
@yang1070 Please find the following updates:

 - ATF: https://github.com/LuxoftSDL/sdl_atf/pull/8/commits/765a9524dd7805c7159fb86fd7ddd54009ca5a7f
 - Scripts: https://github.com/LuxoftSDL/sdl_atf_test_scripts/pull/129/commits/33ac041b9f02979106f563b98b11504cd97a2a4c

These changes allow ATF to create loop device only if it's required by script.
I did test them within new and existing scripts and they work fine.

Even though these changes work pretty stable I would think that loop device possibility requires a separate proposal.

Current solution has a few drawbacks:
 - Pure flexibility. Now it's supported only write rate for a loop device. In case if we need to use other restrictions ATF would require an update. (e.g. drive size, read speed etc.) Though we can quite simply add a few more prospective options in a current implementation.
 - ATF runner (before executing the script) fetches script's content to obtain loop device options ( e.g. `config.isLoopDeviceRequired` and `config.loopDeviceWriteRate` parameters). I guess this is not the best approach and there may be better one.
 - There is a limit on number of available loop devices. It should correlate with the number of parallel workers. Issue would occur when total number of scripts currently assigned to the workers which require loop device exceeds number of available loop devices (in the system). This number is variable and depends on environment.
 - Fault tolerance and error handling needs to be improved for the cases when loop device can't be created.

From other hand current solution is workable and we spent some amount of time to implement it. It might be not a good option to leave it in a specific branch.

We can fix some of the issues and try to push this PR into Open as part of feature implementation. Then to see what Project Maintainers think about it. 

I would appreciate your feedback what do you think would be the best option here.